lib/inspec/profile_vendor.rb

```diff
@@ -80,5 +82,40 @@ def extract_archives
         File.delete(filepath)
       end
     end
+
+    # Check top-level profile metadata for any local archives
+    # if present, go ahead and inflate it into the vendor cache.
+    # This will cause subsequent fetchers to hit on the vendor cache,
+    # avoiding a possibly failing fetch under airgapped conditions.
+    def warm_vendor_cache_from_archives
+      # We need to determine the cwd of the profile, to expand the (likely)
+      # relative path of the dependencies. There is no non-invasive way of doing that.
+      profile_cwd = profile.source_reader.target.parent.path
+
+      # list dependencies in profile
+      profile.metadata.dependencies.each do |dep_info|
+        next unless dep_info.key?(:path) # "Local"-type fetchers are identified by the :path key
+
+        # Locate local dep
+        dep_profile_path = File.expand_path(dep_info[:path], profile_cwd)
+        raise(Inspec::FetcherFailure, "Profile dependency local path '#{dep_profile_path}' does not exist") unless File.exist?(dep_profile_path)
+
+        # Determine if it is an archive
+        file_provider = FileProvider.for_path(dep_profile_path)
+        next unless file_provider.is_a?(ZipProvider) || file_provider.is_a?(TarProvider)
+
+        # Place the local dependency in the vendor cache.
+        # Fetchers are in charge of calculating cache keys.
+        fetcher = Inspec::Fetcher::Local.resolve(dep_profile_path)
+        dep_profile_cache_dir = "#{cache_path}/#{fetcher.cache_key}"
+        file_provider.extract(dep_profile_cache_dir)
+
+        # The archive (probably) contained a vendor cache of its own.
+        # We must flatten that, so that all vendor cache entries are top-level.
+        Dir["#{dep_profile_cache_dir}/vendor/*"].each do |sub_dep_cache_dir|
+          FileUtils.mv(sub_dep_cache_dir, cache_path)
```

Comment:
Collision if two dependencies have the same name and get moved to cache_path

Reply:
That's just going to be a limitation. We'll need to document that.

Reply:
doco doco? or comment here?